modes/mu4e/evil-collection-mu4e.el

```diff
@@ -68,349 +68,209 @@
-                                               'utf-8 t)
-                                              query)
-                                        collect q))
-            for unread = (and qcounts (plist-get (car qcounts) :unread))
-            when (not (plist-get m :hide))
-            when (not (and mu4e-main-hide-fully-read (eq unread 0)))
-            concat (concat
-                    ;; menu entry
-                    (evil-collection-mu4e--main-action-str
-                     (concat "\t* [J" key "] " name)
-                     (concat "J" key))
-                    ;; append all/unread numbers, if available.
-                    (if qcounts
-                        (let ((unread (plist-get (car qcounts) :unread))
-                              (count (plist-get (car qcounts) :count)))
-                          (format
-                           "%s (%s/%s)"
-                           (make-string (- longest (string-width name)) ? )
-                           (propertize (number-to-string unread)
-                                       'face 'mu4e-header-key-face)
-                           count))
-                      "")
-                    "\n"))
-(propertize "\n  Misc" 'face 'mu4e-title-face)))
-
-(defun evil-collection-mu4e-replace-region (new-region start end)
-  "Replace region between START and END with NEW-REGION.
-START end END end are regular expressions."
-  ;; move to start of region
-  (goto-char (point-min))
-  (re-search-forward start)
-
-  ;; insert new headings
-  (insert "\n\n")
-  (insert new-region)
-  ;; Delete text until end of region.
-  (let ((start-point (point))
-        (end-point (re-search-forward end)))
-    (delete-region start-point end-point)))
-
-(defun evil-collection-mu4e-update-main-view ()
-  "Update `Basic', `Maildir', and `Misc' regions to reflect the new keybindings."
-  (evil-collection-mu4e-replace-region evil-collection-mu4e-new-region-basic
-                                       evil-collection-mu4e-begin-region-basic
-                                       evil-collection-mu4e-end-region-basic)
-  (evil-collection-mu4e-replace-region (evil-collection-mu4e-new-region-misc)
-                                       evil-collection-mu4e-begin-region-misc
-                                       evil-collection-mu4e-end-region-misc)
-  (evil-collection-mu4e-replace-region (evil-collection-mu4e-new-region-maildir)
-                                       evil-collection-mu4e-begin-region-maildir
-                                       evil-collection-mu4e-end-region-maildir))
-
-
-;;; Initialize evil-collection-mu4e
-
-;;;###autoload
-(defun evil-collection-mu4e-setup ()
-  "Initialize evil-mu4e if necessary.
-If mu4e-main-mode is in evil-state-motion-modes, initialization
+(if (version< mu4e-mu-version "1.10")
+    (require 'evil-collection-mu4e
+             (expand-file-name "modes/mu4e/evil-collection-mu4e-1.8" evil-collection-base-dir))
```

Comment:
1. [ ] Unit tests seem to break due to `mu4e-mu-version` being a void variable. Not sure how this can be fixed. mu4e needs to be loaded to know its version. We could of course cheat with something like
   ``` elisp
   (if (and (boundp mu4e-mu-version) (version< mu4e-mu-version "1.10")) ...
   ```
   But this feels like not a proper solution. Maybe something with `eval-after-load` or something like that?

2. [ ] I What about users of 1.9, which started the transition to the new interface? In my opinion that's officially not a stable version but a development-version, so I wouldn't load any backwards-compatibility code for that. As a non-stable release, people who used that are likely to be users of bleeding-edge mu4e versions and will update soon.

3. [ ] I would add a comment above that `if` explaining why this is being done